src/main/kotlin/com/avail/descriptor/pojos/PojoFieldDescriptor.kt

```diff
@@ -0,0 +1,334 @@
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+import java.util.*
+
+/**
+ * A `PojoFieldDescriptor` is an Avail [variable][VariableDescriptor] that
+ * facilitates access to the instance [Java field][Field] of a particular
+ * [pojo][PojoDescriptor] or the static field of a particular
+ * [pojo&#32;type][PojoTypeDescriptor]. It supports the same protocol as any
+ * other variable, but reads and writes are of the pojo's field.
+ *
+ * @constructor
+ *
+ * @param mutability
+ *   The [mutability][Mutability] of the new descriptor.
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+class PojoFieldDescriptor private constructor(
+	mutability: Mutability
+) : Descriptor(
+	mutability, TypeTag.VARIABLE_TAG, ObjectSlots::class.java, null) {
+	/** The layout of the object slots.  */
+	enum class ObjectSlots : ObjectSlotsEnum {
+		/**
+		 * A [raw&#32;pojo][RawPojoDescriptor] that wraps a reflected Java
+		 * [Field].
+		 */
+		FIELD,
+
+		/**
+		 * The receiver [raw&#32;pojo][RawPojoDescriptor] to which the [Field]
+		 * is bound.
+		 */
+		RECEIVER,
+
+		/**
+		 * The [kind][VariableTypeDescriptor] of the
+		 * [variable][VariableDescriptor].
+		 */
+		KIND
+	}
+
+	override fun o_ClearValue(self: AvailObject) {
+		val receiver = self.slot(RECEIVER).javaObjectNotNull<Any>()
+		val field = self.slot(FIELD).javaObjectNotNull<Field>()
+		val fieldType = field.type
+		val defaultValue: Any?
+		// Sadly Java does not offer reflective access to the default values of
+		// its primitive types ...
+		defaultValue = when {
+			!fieldType.isPrimitive -> null
+			fieldType == Boolean::class.javaPrimitiveType ->
+				java.lang.Boolean.FALSE
+			fieldType == Float::class.javaPrimitiveType -> 0.0f
+			fieldType == Double::class.javaPrimitiveType -> 0.0
+			fieldType == Char::class.javaPrimitiveType -> 0.toChar()
+			else -> zero().marshalToJava(fieldType)
+		}
+		// Clear the variable by writing the appropriate default value.
+		try {
+			synchronized(receiver) { field[receiver] = defaultValue }
+		} catch (e: Exception) {
+			throw MarshalingException(e)
+		}
+	}
+
+	@AvailMethod
+	override fun o_Equals(self: AvailObject, another: A_BasicObject): Boolean =
+		another.equalsPojoField(self.slot(FIELD), self.slot(RECEIVER))
+
+	@AvailMethod
+	override fun o_EqualsPojoField(
+		self: AvailObject,
+		field: AvailObject,
+		receiver: AvailObject
+	): Boolean = (self.slot(FIELD).equals(field)
+		&& self.slot(RECEIVER).equals(receiver))
+
+	@AvailMethod
+	@Throws(VariableGetException::class)
+	override fun o_GetValue(self: AvailObject): AvailObject {
+		val receiver = self.slot(RECEIVER).javaObjectNotNull<Any>()
+		val field = self.slot(FIELD).javaObjectNotNull<Field>()
+		val expectedType = self.slot(KIND).readType()
+		return try {
+			synchronized(receiver) {
+				PojoTypeDescriptor.unmarshal(field.get(receiver), expectedType)
+			}
+		} catch (e: Exception) {
+			throw AvailRuntimeException(E_JAVA_MARSHALING_FAILED, e)
+		}
```

Comment:
Put open braces on own line

Reply:
Also moved return inside try.  When possible, that usually looks nicer, as it doesn't hide the try by pushing it to the right.